sql/item_jsonfunc.cc

```diff
@@ -4060,7 +4060,7 @@ bool Item_func_json_objectagg::add()
     result.append(", ");
 
   result.append("\"");
-  result.append(*key);
+  result.append_for_single_quote(key->ptr(), key->length());
```

Comment:
Except its not single quotes.

Found `st_append_escaped` that looks suitable. As `StringBuffer` is a subclass of `String`, `&buf` still can be used.

Reply:
Yes, it is a better approach.